core/new-gui/src/app/workspace/component/product-tour/product-tour.component.ts

```diff
@@ -8,7 +8,22 @@ import { TourService, IStepOption } from 'ngx-tour-ng-bootstrap';
 })
 export class ProductTourComponent implements OnInit {
 
-  constructor(public tourService: TourService) { }
+  public loaded = false;
+
+  constructor(public tourService: TourService) {
+    this.tourService.end$.subscribe(() => {
+      location.reload();
+    });
+
+    this.tourService.initialize([{
```

Comment:
Should we initialize the tourService inside the constructor or in other Angular life cycle callbacks, such as `onInit`?

Reply:
In the library's demo project code, the author initialize tourService inside the constructor